Add auto-merge GitHub workflow

Cruikshanks opened the pull request:
We currently have to maintain Dependabot updates across 11 repos for the Water Abstraction Service.

All we do when checking a Dependabot update is confirm it hasn't broken CI. We then approve and merge.

We are not really adding any value to the process. So, to reduce the maintenance load on the team we have agreed that any Dependabot update that passes our CI build can be automatically merged.

This adds [Dependabot Auto Merge](https://github.com/marketplace/actions/dependabot-auto-merge) to the repo which is a GitHub action that should handle this for us.

andrewhick commented:
Minor UI tweaks using existing syntax so not requesting review for this PR.  Merging automatically.

Cruikshanks committed:
Add auto-merge GitHub workflow

We currently have to maintain Dependabot updates across 11 repos for the Water Abstraction Service.

All we do when checking a Dependabot update is confirm it hasn't broken CI. We then approve and merge.

We are not really adding any value to the process. So, to reduce the maintenance load on the team we have agreed that any Dependabot update that passes our CI build can be automatically merged.

This adds [Dependabot Auto Merge](https://github.com/marketplace/actions/dependabot-auto-merge) to the repo which is a GitHub action that should handle this for us.
